java/lang/Float/Float.cpp

```diff
@@ -142,68 +140,483 @@ Float Float::operator/=(const Float &target) const {
     return (Float) (this->original / target.original);
 }
 
-char Float::charValue() const {
-    string convertResult = string_from_float(this->original);
-    char charValueResult = string_to_char(convertResult);
-    free(convertResult);
-    return charValueResult;
-}
-// TODO(thoangminh): enable after finish toString()
-//string Float::stringValue() const {
-//    return (string) this->toString();
-//}
-
 float Float::floatValue() const {
     return this->original;
 }
 
 short Float::shortValue() const {
-    return (short) this->original;
+    return static_cast<short> (this->original);
 }
 
 int Float::intValue() const {
-    return (int) this->original;
+    return static_cast<int> (this->original);
 }
 
-/**
- * Long value
- *
- * @return long
- */
 long Float::longValue() const {
-    return (long) this->original;
+    return static_cast<long> (this->original);
 }
 
-/**
- * Float value
- *
- * @return
- */
 double Float::doubleValue() const {
-    return this->original;
+    return static_cast<double> (this->original);
+}
+
+byte Float::byteValue() const {
+    return static_cast<byte> (this->original);
 }
 
-/**
- * Float to String
- *
- * @return String
- */
 string Float::toString() const {
     return this->originalString;
 }
 
-/**
- * String to Float
- *
- * @param target
- * @return Float
- */
+Float Float::parseFloat(String inputString) {
+    return static_cast<Float> (string_to_float(inputString.toString()));
+}
+
+boolean Float::isFinite(float valueFloat) {
+    return (Math::abs(valueFloat) <= MAX_VALUE);
+}
+
+boolean Float::isInfinite(float valueFloat) {
+    boolean isPOSITIVE_INFINITY = (valueFloat == POSITIVE_INFINITY);

```

Comment:
Please use Float:: before Static field, please check for others too.
Please change isPOSITIVE_INFINITY to isPositiveInfinity

Reply:
Thanks. Fixed.